pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -219,7 +219,7 @@ func (r *ReconcileComplianceRemediation) reconcileRemediation(instance *compv1al
 		return common.NewNonRetriableCtrlError(
 			"Unable to get fix object from ComplianceRemediation. "+
 				"Please update the compliance-operator's permissions: %w", err)
-	} else if runtime.IsNotRegisteredError(err) || meta.IsNoMatchError(err) {
+	} else if runtime.IsNotRegisteredError(err) || meta.IsNoMatchError(err) || (err != nil && strings.Contains(err.Error(), "the server could not find the requested resource")) {
```

Comment:
OK, so this is the ugly workaround. I cannot for the life of me figure out how to catch the proper error. In the error string chain I see `unable to retrieve the complete list of server APIs` which would suggest the error is `ErrGroupDiscoveryFailed` which should be testable for with `IsGroupDiscoveryFailedError` from `k8s.io/client-go/discovery/discovery_client.go` but that doesn't seem to work for me. Which is especially odd with go 1.20 where `errors.Is` should even work across wrapped errors. I'll continue trying to get to the root of the issue, but as the worst case solution, maybe this workaround would at least pass CI.

Reply:
Looks like it worked for CI at least.

Reply:
Strange - after looking at the implementation in `client-go` I'm not sure why we wouldn't be able to use:

```go
IsGroupDiscoveryFailedError(err)
```